React/wbooks/src/app/screens/BookDetail/index.js

```diff
@@ -1,58 +1,28 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom';
-
-import defaultBookIcon from '../../assets/default_book.svg';
-import errorIcon from '../../assets/sad_icon.png';
-import * as bookService from '../../../services/bookService';
-
-import BookSummary from './components/BookSummary/index.js';
-import Suggestion from './components/Suggestion/index.js';
-import NewComment from './components/NewComment/index.js';
-import Comment from './components/Comment/index.js';
-import styles from './styles.scss';
-
-class BookDetail extends Component {
-  state = { book: '', error: '' };
-  backTitle = '<Volver';
-
-  componentWillMount = () => {
-    bookService.getBookDetail(this.props.match.params.bookId)
-    .then((response) => {this.setState({ book: response.data })})
-    .catch((error) => {this.setState({ error: 'Sorry, the book was not found.' })});
-  }
+import { connect } from 'react-redux';
+
+import * as booksActions from '../../../redux/books/actions';
+
+import BookDetail from './layout.js';
+
+class BookDetailContainer extends Component {
+  componentWillMount = () => this.props.getBookDetail(this.props.match.params.bookId);
```

Comment:
Component life-cycles methods should not be arrow functions. Lifecycle methods should be class functions, not instance functions. Write all methods in classes as arrow functions, except for lifecycle methods.